gluejobutils/s3.py

```diff
@@ -1,21 +1,130 @@
+    log_file = StringIO()
+    json.dump(data, log_file, indent=4, separators=(',', ': '))
+    log_obj = s3_resource.Object(bucket, key)
+    log_upload_resp = log_obj.put(Body=log_file.getvalue())
+    return log_upload_resp
+
+def get_filepaths_from_s3_folder(s3_folder_path, parquet = False) :
+    """
+    Get a list of filepaths from a bucket. If parquet is set to True then only return a distinct list of folder
+    paths that contain files with a .parquet extension.
+    """
+    if not isinstance(parquet, bool) :
+        raise TypeError('input param parquet must be logical (True/False)')
+
+    s3_folder_path = add_slash(s3_folder_path)
+    bucket, key = s3_path_to_bucket_key(s3_folder_path)
+
+    s3b = s3_resource.Bucket(bucket)
+    obs = s3b.objects.filter(Prefix = key)
+    ob_keys = [o.key for o in obs]
+    
+    if parquet :
+        ob_keys = list(set(["/".join(o.split('/')[:-1]) + '/' for o in ob_keys if o[-8:] == '.parquet']))
+
+    ob_keys = sorted([bucket_key_to_s3_path(bucket, o) for o in ob_keys])
+
+    return ob_keys
+
+def copy_s3_folder_contents_to_new_folder(old_s3_folder_path, new_s3_folder_path) :
+    old_s3_folder_path = add_slash(old_s3_folder_path)
+    new_s3_folder_path = add_slash(new_s3_folder_path)
+
+    all_old_filepaths = get_filepaths_from_s3_folder(old_s3_folder_path)
+    for ofp in all_old_filepaths :
+        nfp = ofp.replace(old_s3_folder_path, new_s3_folder_path)
+        copy_s3_object(ofp, nfp)
+
+def delete_s3_object(s3_path) :
+    """
+    Deletes the file at the s3_path given.
+    """
+    b, o = s3_path_to_bucket_key(s3_path)
+    s3_resource.Object(b, o).delete()
+
+def delete_s3_folder_contents(s3_folder_path) :
+    """
+    Deletes all files within the s3_folder_path given given.
+    """
+    s3_folder_path = add_slash(s3_folder_path)
+    all_filepaths = get_filepaths_from_s3_folder(s3_folder_path)
+    for f in all_filepaths :
+        delete_s3_object(f)
+
+def copy_s3_object(old_s3_path, new_s3_path) :
+    """
+    Copies a file in S3 from one location to another.
+    """
+    new_bucket, new_key = s3_path_to_bucket_key(new_s3_path)
+    if 's3://' in old_s3_path :
+        old_s3_path = old_s3_path.replace('s3://', '')
+    s3_resource.Object(new_bucket, new_key).copy_from(CopySource=old_s3_path)
+
+# https://stackoverflow.com/questions/33842944/check-if-a-key-exists-in-a-bucket-in-s3-using-boto3
+def check_for_s3_file(s3_path) :
+    """
+    Returns True/False depending if a file exists in the S3 path provided.
+    exists = check_for_s3_file(s3://my-bucket/file.txt)
+    """
+    bucket, key = s3_path_to_bucket_key(s3_path)
+    try:
+        s3_resource.Object(bucket, key).load()
+    except botocore.exceptions.ClientError as e:
+        if e.response['Error']['Code'] == "404":
+            return False
+        else:
+            raise
+    else:
+        # The object does exist.
+        return True
+
+def check_for_parquet_data_in_folder(s3_folder_path) :
```

Comment:
Need to clarify what this does.  Does it check there's ONLY ONE parquet file in folder and no other files?  If so function name should be more specific.

Should this be called 'folder_contains_only_a_single_parquet_file'

Reply:
'folder_contains_only_a_single_parquet_file' is not what happens due to issues raised above so this function is also a bit dodge.